lib/trace-writer.js

```diff
@@ -78,12 +78,16 @@ TraceWriter.prototype.queueTrace_ = function(trace) {
     if (err) { return; } // ignore as index.js takes care of this.
 
     trace.projectId = project;
-    var length = that.buffer_.push(trace);
 
-    that.logger_.info('queued trace. new size:', length);
+    if (that.buffer_.length <= that.config_.bufferSize) {
+      that.buffer_.push(trace);
+      that.logger_.info('queued trace. new size:', that.buffer_.length);
```

Comment:
While you're in here, the log level of INFO on each trace is too aggressive. We should probably drop it down to DEBUG.